src/utils/util.ts

```diff
@@ -231,13 +231,23 @@ export function getDataDictionary(columns: Columns): DataDictionary {
           );
 
           dictionaryEntry.Annotations.Levels = Object.entries(column.levels).reduce(
-            (termsObj, [levelKey, levelValue]) => ({
-              ...termsObj,
-              [levelKey]: {
-                TermURL: levelValue.termURL || '',
-                Label: levelValue.label || '',
-              },
-            }),
+            (termsObj, [levelKey, levelValue]) => {
+              if (levelValue.termURL && levelValue.label) {
+                // Include properly mapped levels
```

Comment:
```suggestion
              if (levelValue.termURL && levelValue.label) {
                // Include the complete levels annotation 
                return {
                  ...termsObj,
                  [levelKey]: {
                    TermURL: levelValue.termURL,
                    Label: levelValue.label,
                  },
                };
              }
              // Replace the incomplete levels annotation with an empty object
              // to avoid a data dictionary with undefined values and to raise a 
              // warning to the user based on the schema validation
              return {
                ...termsObj,
                [levelKey]: {} as { TermURL: string; Label: string },
              };
            },
```
Is that about right? I.e. the reason you don't just get rid of the `|| ''` bit in the old code is not just to appease typescript, but also for UX reasons? Would also be good to point out here if the schema would not trigger if we had `TermURL` as a key but with `undefined` as a value